hi-ml-histopathology/docs/panda_model.md

```diff
@@ -0,0 +1,124 @@
+- What results are expected (accuracy)
+- If people want to re-use that model on their data:
+- What is the expected data format?
+- Which parameters need to be tuned?
+
+## Background
+
+The repository contains the configuration for training Deep Multiple Instance Learning (DeepMIL) models for ISUP score
+prediction on the [PANDA challenge
+dataset](https://www.kaggle.com/c/prostate-cancer-grade-assessment?msclkid=63cd71d8cf6511ec8191222e2876cfec).
+The models are developed to reproduce the results described in
+[Myronenko et al. 2021](<https://link.springer.com/chapter/10.1007/978-3-030-87237-3_32>) and the example in the
+[Project-MONAI
+tutorial](https://github.com/Project-MONAI/tutorials/blob/master/pathology/multiple_instance_learning/panda_mil_train_evaluate_pytorch_gpu.py).
+
+A ResNet50 pre-trained encoder checkpoint is downloaded on-the-fly (from <https://download.pytorch.org/models/>) at the
+start of the training run.
+
+## Preparations
+
+Please follow the instructions in the [Readme file](../README.md#setting-up-python) to create a Conda environment and
+activate it, and the instructions to [set up Azure](../README.md#setting-up-azureml).
+
+## Mount datasets
+
+If you would like to use the models interactively to debug and/or devlop, it is necessary to mount the datasets that are
+available in Azure. Instructions to prepare and upload the PANDA dataset are [here](public_datasets.md).
+"Mounting" here means that the dataset will be loaded on-demand over the network (see also [the
+docs](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-train-with-datasets#mount-vs-download)).
+
+You can mount the dataset by executing this script in `<root>/hi-ml-histopathology`:
+
+```shell
+python src/histopathology/scripts/mount_azure_dataset.py --dataset_id PANDA
+```
+
+After a few seconds, this may bring up a browser to authenticate you in Azure, and let you access the AzureML
+workspace that you chose by downloading the `config.json` file. If you get an error message saying that authencation
+failed, "The token is not yet valid (nbf)", please ensure that your
+system's time is set correctly (on WSL, use `sudo hwclock -s`) and then try again.
+
+Upon success, the script will print out:
+
+```text
+Dataset PANDA will be mounted at /tmp/datasets/PANDA.
+```
+
+## Running the model as-is
+
+If you have a GPU available, you can run training on that machine, by executing in `<root>/hi-ml-histopathology`:
+
+```shell
+conda activate HimlHisto
+python ../hi-ml/src/health_ml/runner.py --model histopathology.SlidesPandaImageNetMILBenchmark
+```
+
+However, the GPU demand for this model is rather high. We recommend running in AzureML, on a GPU compute cluster:
+
+```shell
+conda activate HimlHisto
+python ../hi-ml/src/health_ml/runner.py --model histopathology.SlidesPandaImageNetMILBenchmark --cluster=innereye4cl --additional_env_files environment.yml
```

Comment:
`--is_finetune` needs to be added to enable fine-tuning, which gives the best results (fine-tuning enabled via is_finetune param)